action.yml

```diff
@@ -0,0 +1,126 @@
+name: "Docker Continuous Integration"
+
+description: |
+  Build and test code, push to docker registry and send a deployment event
+
+inputs:
+  app_checkout_path:
+    description: "The path to checkout the application repository to during the workflow run"
+    default: ./app-repo
+    required: false
+  deploy:
+    description: "A `true` value will trigger a GitHub deployment"
+    required: false
+    default: "false"
+  docker_registry:
+    description: "Registry where docker image is stored"
+    required: true
+  image_name:
+    description: "If present will override derived docker image name"
+    required: false
+    default: ${{ github.event.repository.name }}
+  org_access_token:
+    description: |
+      Secret token for accessing other repos in the GitHub Org. The `secrets` context
+      is not available in composite actions and must be passed in from the calling action
+    required: false
+  run_tests:
+    description: "A `true` value will result in running tests"
+    default: "false"
+    required: false
+  secrets: # https://github.com/docker/build-push-action/blob/master/docs/advanced/secrets.md
+    description: "List of secrets to expose to the build (eg. key=string, GIT_AUTH_TOKEN=mytoken)"
+    required: false
+  stages:
+    description: "JSON formatted map of the environments and repos to deploy to based off GitHub event"
+    required: true
+  subdirectory:
+    description: "Subdirectory of docker file"
+    required: false
+    default: "."
+  tests_script:
+    description: "Script to run during tests step, requires `run_tests` to be set to true, script needs to be executable"
+    default: tests.sh
+    required: false
+
+runs:
+  using: composite
+  steps:
+    #####################################
+    # Environment and variable setup
+    - name: Set composite level variables
+      id: vars
+      uses: actions/github-script@v5
+      with:
+        script: |
+          const actionContext = require('./lib/actioncontext.js')(this, context, core, github, ${{ toJSON(github) }}, ${{ toJSON(inputs) }}, ${{ toJSON(steps) }})
+          const deployment = require('./lib/deployment.js')(actionContext)
+
+          actionContext.setOutput("image_build_enabled", deployment.getInfoCI().image_build_enabled)
+          actionContext.setOutput("tests_enabled", "${{ github.event.action }}" != 'closed' && "${{ inputs.run_tests }}" == 'true')
+          actionContext.setOutput("image_tag_sha", `${{ inputs.docker_registry }}/${{ inputs.image_name }}:${{ github.sha }}`)
+
+    - name: Check out repository code
+      if: github.event.action != 'closed'
```

Comment:
If the PR is closed, how come only this step is skipped ? Should other steps also be skipped ?